app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/FavoritesViewModel.kt

```diff
@@ -96,35 +103,58 @@ class FavoritesViewModel @Inject constructor(
                                     it.items?.any { menuItem -> menuItem.name == itemName } == true
                                 } == true
                             }?.description
-                        }
-                            .mapValues { mapEntry ->
-                                mapEntry.value.mapNotNull { eatery -> eatery.name }
-                            }
-                            .mapKeys { (key, _) ->
-                                if (key == null || key !in listOf(
-                                        "Breakfast",
-                                        "Lunch",
-                                        "Dinner"
-                                    )
-                                ) "Other" else key
-                            }
+                        }.mapValues { mapEntry ->
+                            mapEntry.value.mapNotNull { eatery -> eatery.name }
+                        }.mapKeys { (key, _) ->
+                            if (key == null || key !in listOf(
+                                    "Breakfast",
+                                    "Lunch",
+                                    "Late lunch",
+                                    "Brunch",
+                                    "Dinner"
+                                )
+                            ) "Other" else key
+                        }.toSortedMap { s1, s2 -> s1.toSortOrder().compareTo(s2.toSortOrder()) }
                     )
                 }.sortedByDescending { it.availability.statusColor == Green }
 
 
                 FavoritesScreenViewState.Loaded(
-                    eateries = favoriteEateries,
-                    favoriteCards = itemFavoriteCards
+                    eateries = filteredEateries,
+                    favoriteCards = itemFavoriteCards,
+                    selectedItemFilters = selectedItemFilters,
+                    selectedEateryFilters = selectedEateryFilters,
                 )
             }
         }
     }.stateIn(viewModelScope, SharingStarted.Eagerly, FavoritesScreenViewState.Loading)
 
-    fun removeFavorite(eateryId: Int?) {
-        if (eateryId != null) userPreferencesRepository.setFavorite(eateryId, false)
+    private fun String.toSortOrder(): Int = when (this) {
+        "Breakfast" -> 1
+        "Brunch" -> 2
+        "Lunch" -> 3
+        "Late lunch" -> 4
+        "Dinner" -> 5
+        else -> Int.MAX_VALUE
     }
 
-    fun toggleFilter(filter: Filter) {
-        _selectedFiltersFlow.update { if (filter in it) it - filter else it + filter }
+    fun removeFavoriteMenuItem(menuItemName: String) = viewModelScope.launch {
+        userPreferencesRepository.toggleFavoriteMenuItem(menuItemName)
+    }
+
+    fun toggleEateryFilter(filter: Filter.FromEatery) {
+        selectedEateryFiltersFlow.update {
+            if (filter in it) it - filter else it + filter
+        }
+    }
```

Comment:
Hmm same thing here with the locations, we should make it so only north or central or west can be selected.

This seems to be a recurring problem, lol. Maybe we can just make some extension function on `List<Filter>` that will do this culling for us?